salt/pillar/__init__.py

```diff
@@ -302,6 +298,119 @@ def __init__(self, opts, grains, minion_id, saltenv, ext=None, functions=None,
             else:
                 log.error('Pillar data must be a dictionary')
 
+    @property
+    def pillar(self):
+        '''
+        Return the current pillar state.
+        '''
+        # FIXME: should this return a deepcopy rather than a reference?
+        # FIXME: should this merge in self.pillar_override?
+        # FIXME: maybe keep a hash of the accumulator to check if other objects need to regenerate?
+        pillar = copy.deepcopy(self._pillar_accumulator)
```

Comment:
We need to carefully benchmark this. Some pillars are enormous and this might be very slow.